### DISCUSSION TITLE
How to add external data source like mysql, oracle and so on like nocodb or tooljet

### DISCUSSION THREAD

**Discussion by liudonghua123** [directus/directus#20801] [Feature Requests]
### Summary

Hi, I just want to connect to the existing data source and expose REST/GraphQL api according to the tables in it.

### Motivation

Support external data source.

### Detailed Design

N.A.

### Requirements List

N.A.

### Drawbacks

N.A.

### Alternatives

N.A.

### Adoption Strategy

N.A.

**Comment by liudonghua123** [comment]
I found directus do support a lot of [databases](https://directus.io/toolkit/connect), but how can I add external data source from the admin UI. It seems need to hard code the connection in envs or configurations when running the service and some directus related meta tables need to create for running.

It would be convenient and flexiable if it supports manage the data source in the admin UI.

**Reply by tecspda** [reply]
External providers are maintained by individual developers. Directus only supports popular databases, the share of which is more than 5-10%.

**Comment by liudonghua123** [comment]
Are there any updates about this feature? I know some other low code platforms like [tooljet](https://github.com/ToolJet/ToolJet), [budibase](https://github.com/Budibase/budibase) support this.

![image](https://github.com/directus/directus/assets/2276718/b13129e4-635b-456a-8378-9a7b529fda1a)

![image](https://github.com/directus/directus/assets/2276718/b0bf618b-d4e7-4afb-9941-a3e7ab78f3d3)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
